### PR TITLE
speed up kto loss and some refactor

### PR DESCRIPTION
**Pull request by shivam15s**
#### Memory
image
#### Speed
image

## Testing Done

- Hardware Type: <BLANK>
- [ ] run `make test` to ensure correctness
- [ ] run `make checkstyle` to ensure code style
- [ ] run `make test-convergence` to ensure convergence

### CODE REVIEW

File: src/liger_kernel/chunked_loss/kto_loss.py [hunk 1]
```diff
@@ -63,62 +61,38 @@ def preference_loss_fn(
             - chosen_rewards: Reward signals for chosen responses (detached)
             - rejected_rewards: Reward signals for rejected responses (detached)
         """
-        if ref_chosen_logps is None:
-            ref_chosen_logps = torch.tensor(0.0, device=chosen_logps.device)
-        if ref_rejected_logps is None:
-            ref_rejected_logps = torch.tensor(0.0, device=rejected_logps.device)
+        logratios_chunk = average_log_prob_chunk - ref_average_log_prob_chunk
+        multiplier_chunk = torch.where(preference_labels_chunk, 1, -1)
```
Comment (hebiao064) [inline src/liger_kernel/chunked_loss/kto_loss.py:65]: smart!

**Review by hebiao064 — APPROVED** [review]
